recipe/0002-import-appropriate-libsodium-binary-depending-on-sys.patch

```diff
@@ -0,0 +1,57 @@
+From 9df5eb70b6d5ac09e1d6e2fbfea72184614ca3c7 Mon Sep 17 00:00:00 2001
+From: Jack Olivieri <jolivieri@anaconda.com>
+Date: Thu, 26 Oct 2023 11:52:54 +0200
+Subject: [PATCH] import appropriate libsodium binary depending on system
+
+---
+ src/rbcl/_sodium_build.py | 34 ++++++++++++++++++++--------------
+ 1 file changed, 20 insertions(+), 14 deletions(-)
+
+diff --git a/src/rbcl/_sodium_build.py b/src/rbcl/_sodium_build.py
+index fdc04ca..c425751 100644
+--- a/src/rbcl/_sodium_build.py
++++ b/src/rbcl/_sodium_build.py
+@@ -82,20 +82,26 @@ def emit_libsodium():
+     Emit compiled libsodium binary as hexadecimal string in ``_sodium.py`` file.
+     """
+     # Extract path to compiled libsodium binary.
+-    path_libsodium = 'build/temp/lib/libsodium.so' # Default for non-Windows builds.
+-    if sys.platform == 'win32':
+-        path_directory = (
+-            os.environ.get('LIB', None)
+-            if os.environ.get('LIB', None) is not None else
+-            os.path.join(os.environ['WINDIR'], 'System32') # Windows default.
+-        )
+-        path_libsodium = os.path.join(path_directory, 'libsodium.dll')
+-        if not os.path.exists(path_libsodium):
+-            raise EnvironmentError(
+-                'libsodium DLL not found at ' + path_libsodium + ', ' +
+-                'set environment variable $LIB to path of directory ' +
+-                'containing libsodium DLL'
+-            )
++    if sys.platform == 'darwin':
++        path_libsodium = os.path.join(os.environ['SRC_DIR'], 'build/temp/lib/libsodium.dylib') # for macOS

```

Comment:
You can use 'sys.prefix' instead of a custom path. This will avoid a copy of files.
Example: https://github.com/AnacondaRecipes/python-feedstock/blob/master/recipe/patches/0003-Fix-find_library-so-that-it-looks-in-sys.prefix-lib-.patch

Reply:
I was actually meaning to not have to use `os.environ['SRC_DIR']` at all in order to patch the install files as little as possible. Is there any reason why we prefer referencing the 'sys.prefix'?

Reply:
libsodium is in your host environment already. This is accessible at build time and I would expect sys.prefix to point to it. So you wouldn't have to copy from prefix to SRC_DIR at all.
`path_libsodium = os.path.join(sys.prefix, 'lib/libsodium.dylib') # for macOS`

Reply:
yeah, I knew that. If copying is all we want to avoid, I can change that.

Reply:
I think we could even avoid the embedded entirely since we control everything and just load the dynamic library that we ship in libsodium at https://github.com/nthparty/rbcl/blob/2018e3a0059276ce674d307c111eee9d8238e814/src/rbcl/_sodium.tmpl#L31.

Reply:
I did this in the `_build_sodium.py`. It's exactly the same thing as changing this in the `_sodium.tmpl`.

Reply:
I think JC is referring to what I mentioned here:
https://github.com/AnacondaRecipes/rbcl-feedstock/pull/1#pullrequestreview-1701928978
by doing that we can omit libsodium dependency in host section altogether

Reply:
I don't really understand the remark, though. `libsodium` is a `host` dependency and is pulled from our **default** repo. So we are in total control where it comes from. I changed the __build_sodium.py_ script to plug `libsodium` straight from the `_h_env`.

Reply:
I don't even see why we need to do the embedding/bundling like they do. We can just depend on libsodium at tuntime, Create our our own `_sodium.py` which does `_sodium = cdll.LoadLibrary(os.path.join(sys.prefix, 'lib', 'libsodium.dylib'))`

Reply:
I had not opened the upstream code. Indeed JC's suggestion is the correct one.

Reply:
As to why, consider a vulnerability is discovered on libsodium. If all we do is a LoadLibrary and have a run dependency on libsodium, we can just build a new version of libsodium and all is good. Whereas if libsodium is included in this package as a hexstring as it is now, we still have to update libsodium, but also remember to rebuild this package.

Reply:
@cbouss ok, I hear your point about the vulnerability.
I did it the way I did, because generally I like the idea of touching the upstream source as little as possible. If it's out of security concerns, and how to update `libsodium` without having to rebuild `rbcl`, then I can do that.

Reply:
changed it according to suggestion.